### PR TITLE
Feature/7743 add aks to sql network AB#7743

### PR DESCRIPTION
**Pull request by gavinwakeley1**

### CODE REVIEW

File: Azure/Dev/resource_virtual_network.tf [hunk 1]
```diff
@@ -45,13 +45,19 @@ resource "azurerm_subnet" "aks" {
...
+                          ]
 }
 
 resource "azurerm_subnet" "ehub" {
   name                 = "${var.project}-${var.environment}-${var.sub}-ehub"
   resource_group_name  = azurerm_resource_group.vnet.name
   virtual_network_name = azurerm_virtual_network.vnet.name
   address_prefixes     = [var.sub_ehub]
+
+  service_endpoints    = ["Microsoft.Sql"
```
Comment (stuart-bass-cgi) [inline Azure/Dev/resource_virtual_network.tf]: I don't think we need it on eHub (this is likely a mistake on my part from the UI defaults)

File: Azure/Prod/resource_virtual_network.tf [hunk 2]
```diff
@@ -45,13 +45,19 @@ resource "azurerm_subnet" "aks" {
...
+  service_endpoints    = ["Microsoft.Sql"
+                          ]
 }
 
 resource "azurerm_subnet" "ehub" {
   name                 = "${var.project}-${var.environment}-${var.sub}-ehub"
   resource_group_name  = azurerm_resource_group.vnet.name
   virtual_network_name = azurerm_virtual_network.vnet.name
   address_prefixes     = [var.sub_ehub]
+
```
Comment (stuart-bass-cgi) [inline Azure/Prod/resource_virtual_network.tf]: Don't think ehub needs it

File: Azure/Test/resource_virtual_network.tf [hunk 3]
```diff
@@ -45,13 +45,19 @@ resource "azurerm_subnet" "aks" {
...
+                          ]
 }
 
 resource "azurerm_subnet" "ehub" {
   name                 = "${var.project}-${var.environment}-${var.sub}-ehub"
   resource_group_name  = azurerm_resource_group.vnet.name
   virtual_network_name = azurerm_virtual_network.vnet.name
   address_prefixes     = [var.sub_ehub]
+
+  service_endpoints    = ["Microsoft.Sql"
```
Comment (stuart-bass-cgi) [inline Azure/Test/resource_virtual_network.tf]: Don't think ehub needs it

**Review by stuart-bass-bjss — CHANGES_REQUESTED** [review]

File: Azure/Dev/resource_virtual_network.tf (hunk 1 above)
Comment (gavinwakeley1) [reply Azure/Dev/resource_virtual_network.tf]: Removed eHub from config and added missing rule

File: Azure/Prod/resource_virtual_network.tf (hunk 2 above)
Comment (gavinwakeley1) [reply Azure/Prod/resource_virtual_network.tf]: Removed eHub from config and added missing rule

File: Azure/Test/resource_virtual_network.tf (hunk 3 above)
Comment (gavinwakeley1) [reply Azure/Test/resource_virtual_network.tf]: Removed eHub from config and added missing rule

**Review by stuart-bass-bjss — APPROVED** [review]

**Pull request merged**